lib/commands/app-management.js

```diff
@@ -56,34 +56,10 @@ commands.queryAppState = async function queryAppState (appId) {
 commands.activateApp = async function activateApp (appId) {
   this.log.debug(`Activating '${appId}'`);
   const apiLevel = await this.adb.getApiLevel();
-  // Fallback to Monkey in older APIs
-  if (apiLevel < 24) {
-    // The monkey command could raise an issue as https://stackoverflow.com/questions/44860475/how-to-use-the-monkey-command-with-an-android-system-that-doesnt-have-physical
-    // but '--pct-syskeys 0' could cause another background process issue. https://github.com/appium/appium/issues/16941#issuecomment-1129837285
-    const cmd = ['monkey',
-      '-p', appId,
-      '-c', 'android.intent.category.LAUNCHER',
-      '1'];
-    let output = '';
-    try {
-      output = await this.adb.shell(cmd);
-      this.log.debug(`Command stdout: ${output}`);
-    } catch (e) {
-      this.log.errorAndThrow(`Cannot activate '${appId}'. Original error: ${e.message}`);
-    }
-    if (output.includes('monkey aborted')) {
-      this.log.errorAndThrow(`Cannot activate '${appId}'. Are you sure it is installed?`);
-    }
-    return;
-  }
-
   const stdout = await this.adb.shell([
     'am', (apiLevel < 26) ? 'start' : 'start-activity',
-    '-a', 'android.intent.action.MAIN',
-    '-c', 'android.intent.category.LAUNCHER',
-    // FLAG_ACTIVITY_REORDER_TO_FRONT | FLAG_ACTIVITY_BROUGHT_TO_FRONT | FLAG_ACTIVITY_NEW_TASK
-    '-f', '0x10420000',
-    '-n', await this.adb.resolveLaunchableActivity(appId),
+    '-p', appId,
```

Comment:
was that properly tested? I'm not quite sure about removing flags. Also, does `am` resolve the main app activity properly in such case if we only provide the package id?

Reply:
Yes, I tested a few apps on Android 11. Will do on Android 6 later.
But let me test more about the launchable activities thing...

```
 % adb shell am start-activity -W -p com.google.android.apps.maps -c android.intent.category.LAUNCHER
Starting: Intent { cat=[android.intent.category.LAUNCHER] pkg=com.google.android.apps.maps }
Warning: Activity not started, its current task has been brought to the front
Status: ok
LaunchState: HOT
Activity: com.google.android.apps.maps/com.google.android.maps.MapsActivity
TotalTime: 393
WaitTime: 398
Complete
```

Reply:
mm, it failed to find activities in some apps.
It won't work as same.

Let me test the resume behavior a bit more...

Reply:
I would rather consider adding support of older OS versions to `resolveLaunchableActivity` in appium-adb and removing the monkey fallback completely if we want to do that "right"

Reply:
It seems like without `'-f', '0x10420000',` works as resume.

Reply:
Probably `-f 0x01000000`? https://developer.android.com/reference/android/content/Intent.html#FLAG_ACTIVITY_PREVIOUS_IS_TOP (just a note)